### PR TITLE
chore(integration-tests): Go through axum for engine-v2 integration-tests

### PR DESCRIPTION
**Pull request by Finistere**
Allows us to also test `engine_v2_axum` and tower layers

### CODE REVIEW

File: engine/crates/integration-tests/src/federation/builder.rs [hunk 1]
```diff
@@ -118,201 +103,23 @@ impl EngineV2Builder {
...
-            .into_iter()
-            .map(|subgraph| match subgraph {
-                Subgraph::Mock { type_id, server } => Ok((type_id, server)),
-                Subgraph::Docker { subgraph, .. } => Err(subgraph),
-            })
-            .partition_result();
-
-        TestEngineV2 {
-            engine: Arc::new(engine),
+    pub async fn build(self) -> TestGateway {
```
Comment (Finistere) [inline engine/crates/integration-tests/src/federation/builder.rs:106]: Split the build into different modules

File: engine/crates/integration-tests/src/federation/subgraph.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
+use std::{any::TypeId, borrow::Cow, collections::HashSet};
```
Comment (Finistere) [inline engine/crates/integration-tests/src/federation/subgraph.rs:1]: mostly a copy from existing build

File: engine/crates/integration-tests/src/federation/builder/engine.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
+use std::{collections::HashMap, sync::Arc};
```
Comment (Finistere) [inline engine/crates/integration-tests/src/federation/builder/engine.rs:1]: mostly a copy from existing build

File: engine/crates/integration-tests/src/federation/subgraph.rs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                    .unwrap()
+                    .error_for_status()
+                    .unwrap()
+                    .json::<Response>()
+                    .await
+                    .unwrap()
+                    .data
+                    .into_schema()
+                    .unwrap()
+                    .to_sdl();
```
Comment (pimeys) [inline engine/crates/integration-tests/src/federation/subgraph.rs:68]: ah, good old reqwest and the amount of unwraps!

**Review by pimeys — APPROVED** [review]

**Pull request merged**